STYLE_GUIDE.md

```diff
@@ -0,0 +1,156 @@
+    * **Format:** `<type>(<scope>): <description>`
+    * **Example:** `feat(api): implement oscal validation logic`
+
+---
+
+## 4. Coding Standards
+
+### Go (e.g., `complyctl`)
+
+#### General Guidelines
+
+- **File Naming**: Use lowercase letters and underscores for file names (e.g., `my_file.go`).
+- **Package Names**: Use short, concise, and lowercase names for packages. Avoid underscores and mixed caps.
+- **Error Handling**: Always check for errors and handle them appropriately. Return errors to the caller when necessary.
+- **Testing**: Write tests for your code. Use descriptive names for test functions and include edge cases.
+- **Line Length**: Limit lines to 99 characters when reasonable to improve readability.
+
+#### Licensing and File Headers
+
+```go
+// SPDX-License-Identifier: Apache-2.0
+```
+
+#### Code Formatting
+
+- **Imports**: Use [`goimports`](https://pkg.go.dev/golang.org/x/tools/cmd/goimports) to format and organize imports. Separate standard library, third-party, and project-specific imports with blank lines. Also ensure only necessary imports.
+- **Indentation and Spacing**: Use the [`go fmt`](https://go.dev/blog/gofmt) tool to automatically format your code. This ensures consistent indentation with tabs and proper spacing after commas, colons, and semicolons.
+- **Braces**: Place opening braces on the same line as the statement (e.g., `if`, `for`, `func`).
+
+#### Additional Guidelines
+
+- **Empty Line at End of File**: Ensure that all files include an empty line at the end. This helps with version control diffs and adheres to POSIX standards.
+- Other [Go checks](https://github.com/complytime/complyctl/blob/main/.golangci.yml) are present in CI/CD and therefore it may be useful to also run them locally before submitting a PR.
+- The pre-commit and pre-push hooks can be configured by installing [pre-commit](https://pre-commit.com/) and running `make dev-setup`
+- Complyctl leverages the [charmbracelet/log](https://github.com/charmbracelet/log) library for logging all command and plugin activity. By default, this output is printed to stdout.
+
+
+### Python (e.g., `complyscribe`)
+
+#### General Guidelines
+
+- **Testing**: Write tests for your code. Use descriptive names for test functions and include edge cases.
+- **Line Length**: Limit lines to 99 characters when reasonable to improve readability.
+- **Type Hinting:** Use Python type hints to improve readability and tooling support.
+
+#### Licensing and File Headers
+
+```python
+# SPDX-License-Identifier: Apache-2.0
+```
+
+#### Code Formatting
+
+- **Style:** Uses `black` and `isort` for formatting.
+- **Lint:** Use `flake8` for linting.
+- **Static type check:** Use `mypy` as static type checker.
+- **Non-Python files:** Use [Megalinter](https://github.com/oxsecurity/megalinter) to lint in a CI task. See [megalinter.yaml](https://github.com/complytime/complyscribe/blob/main/.mega-linter.yml) for more information.
+
+#### Additional Guidelines
```

Comment:
This is similar to the Additional Guidelines for `Go`. Maybe we could make a more generic section applicable to everything and treat only exceptions for each language in a shorter separate section. WDYT?